frontend/coprs_frontend/coprs/views/backend_ns/backend_general.py

```diff
@@ -286,16 +286,34 @@ def build_task_canceled(task_id):
 @backend_ns.route("/pending-actions/")
 def pending_actions():
     'get the list of actions backand should take care of'
+    waiting_actions = actions_logic.ActionsLogic.get_waiting()
+    busy_namespaces = set()
+    for action in waiting_actions:
+        if (
+            action.object_type == ActionObjectTypeEnum.COPR
+            and action.action_type == ActionTypeEnum("delete")
+        ):
+            busy_namespaces.add(action.copr.full_name)
+
     data = []
-    for action in actions_logic.ActionsLogic.get_waiting():
+    for action in waiting_actions:
```

Comment:
Are the waiting repos ordered?  If yes, you can just do everything in one cycle:
```python
for action in waiting_actions:
    if "create" or "fork":
        if action.project_name in busy_namespaces:
            continue
    if "delete":
        busy_namespaces.add(action.project_name)
    data.append({...})
```